src/components/views/rooms/MemberInfo.js

```diff
@@ -356,6 +359,73 @@ module.exports = createReactClass({
         });
     },
 
+    onRedactAllMessages: async function() {
+        const {roomId, userId} = this.props.member;
+        const room = this.context.matrixClient.getRoom(roomId);
+        if (!room) {
+            return;
+        }
+        let timeline = room.getLiveTimeline();
+        let eventsToRedact = [];
+        while (timeline) {
+            eventsToRedact = timeline.getEvents().reduce((events, event) => {
+                if (event.getSender() === userId && !event.isRedacted()) {
+                    return events.concat(event);
+                } else {
+                    return events;
+                }
+            }, eventsToRedact);
+            timeline = timeline.getNeighbouringTimeline(EventTimeline.BACKWARDS);
+        }
+
+        const count = eventsToRedact.length;
+        const user = this.props.member.name;
+
+        if (count === 0) {
+            const InfoDialog = sdk.getComponent("dialogs.InfoDialog");
+            Modal.createTrackedDialog('No user messages found to remove', '', InfoDialog, {
+                title: _t("No recent messages by %(user)s found", {user}),
+                description:
+                    <div>
+                        <p>{ _t("Try scrolling up in the timeline to see if there are any earlier ones.") }</p>
+                    </div>,
+            });
+        } else {
+            const QuestionDialog = sdk.getComponent("dialogs.QuestionDialog");
+            const confirmed = await new Promise((resolve) => {
+                Modal.createTrackedDialog('Remove recent messages by user', '', QuestionDialog, {
+                    title: _t("Remove recent messages by %(user)s", {user}),
+                    description:
+                        <div>
+                            <p>{ _t("You are about to remove %(count)s messages by %(user)s. This cannot be undone. Do you wish to continue?", {count, user}) }</p>
+                            <p>{ _t("For large amount of messages, this might take some time. Please don't refresh your client in the meantime.") }</p>
+                        </div>,
+                    button: _t("Remove %(count)s messages", {count}),
+                    onFinished: resolve,
+                });
+            });
+
+            if (!confirmed) {
+                return;
+            }
+
+            // Submitting a large number of redactions freezes the UI,
+            // so first wait 200ms to allow to rerender after closing the dialog.
+            await new Promise(resolve => setTimeout(resolve, 200));
```

Comment:
Maybe some batching as mentioned below would make this not necessary?

Reply:
What makes the UI lock up is en-queuing the events, not sending them. The promise resolves once the event is sent. So if we'd batch them, we'd just add more "yield"s like this to break up the sync queueing code I think...

Having said that, I can probably replace the timeout with something less ugly.

Reply:
Aha, okay... that's a bit depressing that enqueueing takes so long... 😅

Is the jank improved by removing the `await` on the map operation below? As in, we don't really need to block the whole app, since we can just let things happen in the background.

Reply:
Just had a look, and it's mainly ~30ms for every redacted event to relayout the timeline. I've seen layout on the timeline taking that amount of time as well before, need to investigate at some point.

Reply:
Could you please file an issue about this (since you have context) with the `performance` label? Then whenever we do get some breathing room for perf investigation, we'll remember to look at this.